### PR TITLE
Sgrinwis/initialsetup

### PR DESCRIPTION
**Pull request by sgrinwis-d2l**

### CODE REVIEW

File: src/components/awards-leaderboard-ui.js [hunk 1]
```diff
@@ -12,28 +12,159 @@
...
+		</d2l-list>`;
+	}
+
+	firstUpdated() {
+		this._getLeaderboard();
+	}
+
+	async _getLeaderboard() {
+		const myLeaderboard = await LeaderboardService.getLeaderboard(this.orgUnitId);
+		if (myLeaderboard === undefined) {
```
Comment (cbelan-d2l) [inline src/components/awards-leaderboard-ui.js]: At a point where we can remove these?
Or do you want them in there still?

File: src/components/awards-leaderboard-ui.js [hunk 2]
```diff
@@ -12,28 +12,159 @@
...
+			console.log(myLeaderboard); // eslint-disable-line no-console
+		}
+		this.sortedLeaderboardArray = myLeaderboard.Objects.slice(0, 10);
+	}
+
+	createAwardEntry(item) {
+		return html`
+		<d2l-list-item>
+			<div class='awardRow' id="${item.UserId}_Expand" @click="${this.expandClicked}">
+				<div class="awardRank" ?topRank="${item.Rank > 4}">${item.Rank}</div>
```
Comment (cbelan-d2l) [inline src/components/awards-leaderboard-ui.js]: Maybe a const for the 4?

File: src/components/awards-leaderboard-ui.js [hunk 3]
```diff
@@ -12,28 +12,159 @@
...
+					aria-hidden="true">
+				</d2l-profile-image>
+				<div class='creditCount '>
+					<div class='d2l-body-compact'>${item.DisplayName}</div>
+					<div class='d2l-body-small'>${this.getAwardText(item)}</div>
+				</div>
+				<img id="${item.UserId}_ExpandIcon" class="expandButton"  text="Expand" src="${this.fullURLExpand.toString()}"></img>
+			</div>
+			<div id="${item.UserId}_ExpandPanel" class="panel"> 
+				Slide Out Panel!
```
Comment (cbelan-d2l) [inline src/components/awards-leaderboard-ui.js]: Lol - I like this :)

File: src/components/awards-leaderboard-ui.js [hunk 4]
```diff
@@ -12,28 +12,159 @@
...
+		if (item.TotalAwardCount === 0) {
+			return this.localize('awards.none');
+		} else if (item.TotalAwardCount === 0) {
+			return this.localize('awards.one');
+		} else
+			return this.localize('awards.many', {numawards:`${item.TotalAwardCount}`});
+	}
+
+	expandClicked(event) {
+		const panel = this.shadowRoot.getElementById(`${event.target.id  }Panel`);
```
Comment (cbelan-d2l) [inline src/components/awards-leaderboard-ui.js]: Nitpick: `${event.target.id  }`
Remove those two extra spaces

File: src/helpers/leaderboardRoutes.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export const BasVersion = 1.1;
+
+export class LeaderboardRoutes {
+	static ClasslistLeaderboard(orgUnitId) { return `/d2l/api/bas/${BasVersion}/orgunits/${orgUnitId}/classlist/`; }
+	static ProfileImage(userId) {return `/d2l/api/hm/users/${userId}`;}
```
Comment (cbelan-d2l) [inline src/helpers/leaderboardRoutes.js]: Hmm.... looks like the linter isn't running?
Unless if they aren't rules, but spaces before/after the curly braces

File: src/components/awards-leaderboard-ui.js (hunk 1 above)
Comment (sgrinwis-d2l) [reply src/components/awards-leaderboard-ui.js]: At the very least, the one is still quite handy.  I can remove the other

File: src/helpers/leaderboardRoutes.js (hunk 5 above)
Comment (sgrinwis-d2l) [reply src/helpers/leaderboardRoutes.js]: This passed the linter before pushing...

File: src/components/awards-leaderboard-ui.js (hunk 3 above)
Comment (sgrinwis-d2l) [reply src/components/awards-leaderboard-ui.js]: I love my job... :)

File: src/components/awards-leaderboard-ui.js (hunk 2 above)
Comment (sgrinwis-d2l) [reply src/components/awards-leaderboard-ui.js]: Done.

File: src/constants/constants.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export const TopStyleLimit = 3;
```
Comment (cbelan-d2l) [inline src/constants/constants.js]: I like this one better :)

File: src/components/awards-leaderboard-ui.js [hunk 7]
```diff
@@ -12,28 +12,156 @@
...
+
+	expandClicked(event) {
+		const panel = this.shadowRoot.getElementById(`${event.target.id}Panel`);
+		const icon = this.shadowRoot.getElementById(`${event.target.id}Icon`);
+		if (panel.style.maxHeight) {
+			panel.style.maxHeight = null;
+			icon.classList.remove('expandButtonRotated');
+			icon.src = this.fullURLExpand.toString();
+		} else {
+			panel.style.maxHeight = `${panel.scrollHeight  }px`;
```
Comment (cbelan-d2l) [inline src/components/awards-leaderboard-ui.js]: Last nitpick - More extra spaces `${panel.scrollHeight  }`

**Pull request merged**